pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -156,7 +160,9 @@ let
     "--libsubdir=\\$abi/\\$libname"
     (optionalString enableSeparateDataOutput "--datadir=$data/share/${ghc.name}")
     (optionalString enableSeparateDocOutput "--docdir=${docdir "$doc"}")
+  ] ++ optionals stdenv.hasCC [
```

Comment:
This is a little weird. Isn't --with-gcc always needed for GHC?

Reply:
I suppose but those "cross flags" do that. We should really just combine the list of flags and make cross and native more similar.